### ISSUE TITLE
Feature request: Partial form data matching

### ISSUE THREAD

**Issue by nabsofken** [httptoolkit/mockttp#136]
Current the withForm method does as an exact match. I would like to perform partial matching. Is this currently possible? If not, it would be great to have this

**Comment by pimterry** [comment]
I don't think this is correct - currently the form data is partially matched (i.e. all data from `withForm` is required for a match, but extra fields in the request don't matter).

Do you have a clear case where that's not working as you're expecting?
